VoodooI2CHID/VoodooI2CTouchscreenHIDEventDriver.cpp

```diff
@@ -25,67 +25,52 @@ OSDefineMetaClassAndStructors(VoodooI2CTouchscreenHIDEventDriver, VoodooI2CMulti
 // Override of VoodooI2CMultitouchHIDEventDriver
 
 bool VoodooI2CTouchscreenHIDEventDriver::checkFingerTouch(AbsoluteTime timestamp, VoodooI2CMultitouchEvent event) {
-    bool got_transducer = false;
+    absolutetime_to_nanoseconds(timestamp, &last_interaction);
     
-    // If there is a finger touch event, decide if it is single or multitouch.
+    // Avoid phantom clicks happening after scrolling
+    if ((last_interaction - last_multitouch_interaction) < MULTITOUCH_TIMEOUT) {
+        return false;
+    }
+
+    VoodooI2CDigitiserTransducer* transducer = OSDynamicCast(VoodooI2CDigitiserTransducer, event.transducers->getObject(0));
```

Comment:
Is it ever possible that the fetched transducer will be `null` here? Previous code assumed that possibility.

Reply:
It seems to be due to the +1 in the for loop.